.github/workflows/test.yml

```diff
@@ -1,18 +1,30 @@
 name: Test
 
-on: [push, pull_request]
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref_name }}
+  cancel-in-progress: true
+
+env:
+  CARGO_TERM_COLOR: always
 
 jobs:
   build-test:
-    name: Build + Test
+    name: Build & Test
+
+    runs-on: ubuntu-latest
 
     strategy:
       fail-fast: false
       matrix:
         rust:
-          - 1.57.0
-          - stable
-          - nightly
+          - { version: 1.57.0, msrv: true }
+          - { version: stable, msrv: false }
+          - { version: nightly, msrv: false }
```

Comment:
Couldn't you have the msrv on an include?
something like:
```yml
rust: 
  - stable
  - nightly
include:
  - rust: 1.57.0
    msrv: true
```

or
```yml
rust: 
  - stable
  - nightly
  - 1.57.0
include:
  - rust: 1.57.0
    msrv: true
```

This seems to be almost the same usecase: https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#example-expanding-configurations

Reply:
Lets try.

Reply:
Okay, unfortunately it doesn't seem to work because then it doesn't generate jobs for all the different features in the matrix.

Reply:
You can see that here: https://github.com/ModProg/derive-where/actions/runs/7208420043.
Ignoring the error (which is just me forgetting to adjust some more stuff in `no_std`), you can see it only generates a single job for v1.57.0.